core/src/gas.rs

```diff
@@ -301,6 +310,24 @@ pub struct GasLeft {
     pub allowance: u64,
 }
 
+impl From<(u64, u64)> for GasLeft {
+    fn from(value: (u64, u64)) -> Self {
+        Self {
+            gas: value.0,
+            allowance: value.1,
+        }
+    }
+}
+
+impl From<(i64, i64)> for GasLeft {
+    fn from(value: (i64, i64)) -> Self {
+        Self {
+            gas: value.0 as u64,
+            allowance: value.1 as u64,
+        }
```

Comment:
It is more readable to destructure the param.
```suggestion
    fn from((gas, allowance): (i64, i64)) -> Self {
        Self {
            gas: gas as u64,
            allowance: allowance as u64,
        }
```

Reply:
Done